scipy/sparse/linalg/_isolve/iterative.py

```diff
@@ -567,7 +568,7 @@ def cgs(A, b, x0=None, tol=None, maxiter=None, M=None, callback=None,
         return postprocess(x), maxiter
 
 
-def gmres(A, b, x0=None, tol=None, restart=None, maxiter=None, M=None,
+def gmres(A, b, x0=None, tol=_NoValue, restart=None, maxiter=None, M=None,
```

Comment:
I believe the `restrt` will also need bumping to 1.14.0 as well and for completeness we may as well change its default `_NoValue` too

Reply:
Agreed, though for me that doesn't _necessarily_ fall in the purview of https://github.com/scipy/scipy/pull/18488 or this PR.

Reply:
While we are at it, indeed I've added to the deprecation cycle.